editor/utils/dom.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Check whether the selection touches an edge of the container
+ *
+ * @param  {Element} container DOM Element
+ * @param  {Boolean} start     Reverse means check if it touches the start of the container
+ * @return {Boolean}           Is Edge or not
+ */
+export function isEdge( container, start = false ) {
+	if ( [ 'INPUT', 'TEXTAREA' ].indexOf( container.tagName ) !== -1 ) {
+		if ( container.selectionStart !== container.selectionEnd ) {
+			return false;
+		}
+
+		if ( start ) {
+			return container.selectionStart === 0;
+		}
+
+		return container.value.length === container.selectionStart;
+	}
+
+	if ( ! container.isContentEditable ) {
+		return true;
+	}
+
+	const selection = window.getSelection();
+	const range = selection.rangeCount ? selection.getRangeAt( 0 ) : null;
+	const position = start ? 'start' : 'end';
+	const order = start ? 'first' : 'last';
+	const offset = range[ `${ position }Offset` ];
+
+	let node = range.startContainer;
+
+	if ( ! range || ! range.collapsed ) {
+		return false;
+	}
+
+	if ( start && offset !== 0 ) {
+		return false;
+	}
+
+	if ( ! start && offset !== node.textContent.length ) {
+		return false;
+	}
+
+	while ( node !== container ) {
+		const parentNode = node.parentNode;
+
+		if ( parentNode[ `${ order }Child` ] !== node ) {
+			return false;
+		}
+
+		node = parentNode;
+	}
+
+	return true;
+}
+
+/**
+ * Places the caret at start or end of a given element
+ *
+ * @param  {Element} container DOM Element
+ * @param  {Boolean} start     Position: Start or end of the element
+ */
+export function placeCaretAtEdge( container, start = false ) {
+	const isInputOrTextarea = [ 'INPUT', 'TEXTAREA' ].indexOf( container.tagName ) !== -1;
+	if ( isInputOrTextarea ) {
+		container.focus();
+		setTimeout( () => {
+			if ( start ) {
+				container.selectionStart = 0;
```

Comment:
Do we know that `container` still exists at this point? To the more pertinent point, do we need the `setTimeout` ? Can we comment as such to why it is necessary?

Reply:
Yes, it was necessary at the time of writing (not sure why?), but I made some tweaks later, I'll check again.

Reply:
Cool, the timeout is not necessary anymore :)